Mustache - add support for dotted names

dgolda opened the pull request:
Mustache specification defines "dot notation" or "dotted names" for example:
```
{{person.name}}
```
see https://github.com/mustache/spec/blob/master/specs/interpolation.yml#L142-L198

This PR adds to scalate mustache support for dotted names.
It is implemented in Scope trait plus unit tests in DottedNamesTest (based on examples from mustache specification)

magnolia-k commented:
@dgolda 
thanks for PR!

An error is occurring in the CI test.
Could you please check?

dgolda commented:
Yes, I've fixed and now "sbt SetScala213 test unidoc" and others are passing on my PC.
I've found problems with sections having dotted names and context precedence in variables with dotted names, so I had to rewrite implementation. Main change is differentiation whether variable exists but has no value or variable doesn't exist.
So I've added method hasVariable to Scope (and had to exclude it in mima) and VariableResult trait (because None is not sufficient).
